src/Features/CSharp/Portable/ConvertAnonymousTypeToTuple/CSharpConvertAnonymousTypeToTupleDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertAnonymousTypeToTuple
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class CSharpConvertAnonymousTypeToTupleCodeRefactoringProvider : AbstractCodeStyleDiagnosticAnalyzer
+    {
+        public CSharpConvertAnonymousTypeToTupleCodeRefactoringProvider()
+            : base(IDEDiagnosticIds.ConvertAnonymousTypeToTupleDiagnosticId,
+                   new LocalizableResourceString(nameof(FeaturesResources.Convert_anonymous_type_to_tuple), FeaturesResources.ResourceManager, typeof(FeaturesResources)),
+                   new LocalizableResourceString(nameof(FeaturesResources.Convert_anonymous_type_to_tuple), FeaturesResources.ResourceManager, typeof(FeaturesResources)))
+        {
+        }
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+            => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
```

Comment:
Does this analyzer need semantic information, or can it proceed just on syntax?

Reply:
Looks like it can just be syntax!

Reply:
not really... we should still protect against conversions to `object` and `dynamic`

Reply:
that's for the fix.  The analysis portion can be syntax only.